aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -973,6 +973,81 @@ void xlog1py_kernel(TensorIteratorBase& iter) {
   });
 }
 
+void betainc_kernel(TensorIteratorBase& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND2(kBFloat16, kHalf, iter.common_dtype(), "betainc_cpu", [&]() {
+    cpu_kernel(iter, [](scalar_t x, scalar_t a, scalar_t b) -> scalar_t {
+      if (at::_isnan(x)){
```

Comment:
Would be great if you can factor this lambda into `Math.h` and also stick `C10_HOST_DEVICE` to it, so that we don't have to duplicate this for CPU and CUDA.

https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Math.h

You can refer this file to look for `C10_HOST_DEVICE` example (part of WIP PR): 
https://github.com/pytorch/pytorch/blob/29be4bbf32fee23d3e815cfd192ea44ea3857dc1/aten/src/ATen/native/Math.h#L1304

Reply:
Sure, I just did this and works!